Enable all ciphers by default
--- a/ssl/ssl.h   2021-08-24 21:42:11.601832000 +0000
+++ b/ssl/ssl.h   2021-08-24 21:42:28.844893000 +0000
@@ -338,7 +338,7 @@
  * The following cipher list is used by default. It also is substituted when
  * an application-defined cipher list string starts with 'DEFAULT'.
  */
-# define SSL_DEFAULT_CIPHER_LIST "ALL:!EXPORT:!LOW:!aNULL:!eNULL:!SSLv2"
+# define SSL_DEFAULT_CIPHER_LIST "ALL:EXPORT:LOW:NULL:SSLv2"
 /*
  * As of OpenSSL 1.0.0, ssl_create_cipher_list() in ssl/ssl_ciph.c always
  * starts with a reasonable order, and all we have to do for DEFAULT is
