TODO[yo] reenanle me
--- a/ssl/s23_clnt.c	(revision 6f7cd63163f5ff1cbdc489718c72938701c2cdfe)
+++ b/ssl/s23_clnt.c	(date 1576846961000)
@@ -366,6 +366,8 @@
             ssl2_compat = 0;
         if (s->tlsext_status_type != -1)
             ssl2_compat = 0;
+		if (!(s->ctx->options & SSL_OP_ALLOW_UNSAFE_LEGACY_RENEGOTIATION))
+			ssl2_compat = 0;            
 # ifdef TLSEXT_TYPE_opaque_prf_input
         if (s->ctx->tlsext_opaque_prf_input_callback != 0
             || s->tlsext_opaque_prf_input != NULL)
--- a/ssl/s23_srvr.c	(revision 6f7cd63163f5ff1cbdc489718c72938701c2cdfe)
+++ b/ssl/s23_srvr.c	(date 1576846961000)
@@ -549,6 +549,11 @@
         SSLerr(SSL_F_SSL23_GET_CLIENT_HELLO, SSL_R_UNSUPPORTED_PROTOCOL);
         goto err;
 #else
+		if (!(s->ctx->options & SSL_OP_ALLOW_UNSAFE_LEGACY_RENEGOTIATION))
+			{
+			SSLerr(SSL_F_SSL23_GET_CLIENT_HELLO,SSL_R_UNSAFE_LEGACY_RENEGOTIATION_DISABLED);
+			goto err;
+			}
         /* we are talking sslv2 */
         /*
          * we need to clean up the SSLv3/TLSv1 setup and put in the sslv2
