--- a/apps/openssl.cnf     2021-08-25 18:16:21.512939000 +0000
+++ b/apps/openssl.cnf     2021-08-25 18:14:35.481536000 +0000
@@ -18,6 +18,18 @@
 # extensions           =
 # (Alternatively, use a configuration file that has only
 # X.509v3 extensions in its main [= default] section.)
+openssl_conf = openssl_def
+[openssl_def]
+engines = engine_section
+
+[engine_section]
+gost = gost_section
+
+[gost_section]
+engine_id = gost
+dynamic_path = /usr/local/ssl/lib/libgost.so
+default_algorithms = ALL
+CRYPT_PARAMS = id-Gost28147-89-CryptoPro-A-ParamSet
